### PR TITLE
Adding Extraction of Numbers

### PR DESCRIPTION
**Pull request by ProsperousHeart**
Converted @penrods's PHP script into Python in order to add the ability to extract numbers such as:
- "one third of a cup"
- "1 and 3/4 cups"
- "one cup and a half"
- "three quarters cups"

### CODE REVIEW

**Comment by aatchison** [comment]
Thanks!

**Comment by aatchison** [comment]
It looks like you are failing pep8

**Comment by ProsperousHeart** [comment]
Correct, but not sure how. @penrods  and I thought it may be some of the items that had returns, because it could be None or something else.

**Comment by forslund** [comment]
Hi @ProsperousHeart, you can see the list of pep8 issues here: https://travis-ci.org/MycroftAI/mycroft-core/builds/236202695#L5007

The command line tool `pep8` is installed when the mycroft environment is set up so you can check for issues before committing the fix by running `pep8 PATH_TO_FILE`.

And also a big thank you for the work you've done (complete with tests!) =)

**Comment by ProsperousHeart** [comment]
Thanks @forslund - I've been able to do so with help @penrods ... I think I may have the hang of it now (I hope) so I can do more. :) Lastest push with PEP8 done.

**Comment by ProsperousHeart** [comment]
Not sure what happened, but looks like my response was removed?

PEP8 is not installed with mycroft environment - I had to install it myself.

I ran it and it was clean, I ran the test_parse.py script and it was clean ... but now it's saying my pull request failed again?

**Comment by forslund** [comment]
Seems that way =(

On the bright side, all but one issue has been fixed!

The only thing needed is to make remove the extra blank line in `test_parse.py` (Line 43)

File: test/util/test_parse.py [hunk 1]
```diff
@@ -17,6 +18,29 @@ def test_articles(self):
...
+        self.assertEqual(extractnumber("2/3 cups"), 2.0/3.0)
+        self.assertEqual(extractnumber("3/4 cups"), 3.0/4.0)
+        self.assertEqual(extractnumber("1 and 3/4 cups"), 1.75)
+        self.assertEqual(extractnumber("1 cup and a half"), 1.5)
+        self.assertEqual(extractnumber("one cup and a half"), 1.5)
+        self.assertEqual(extractnumber("one and a half cups"), 1.5)
+        self.assertEqual(extractnumber("one and one half cups"), 1.5)
+        self.assertEqual(extractnumber("three quarter cups"), 3.0/4.0)
+        self.assertEqual(extractnumber("three quarters cups"), 3.0/4.0)
+
```
Comment (forslund) [inline test/util/test_parse.py]: Here there should be only one blank line

Comment (ProsperousHeart) [inline test/util/test_parse.py]: Fixed this - thanks. Didn't run pep8 on the test file.  Forgot about that one - didn't run PEP8 on it.

**Pull request merged**